webpack.config.js

```diff
@@ -0,0 +1,37 @@
+const buildValidations = require('./build-utils/build-validations')
+const commonConfig = require('./build-utils/webpack.common')
+
+const { merge } = require('webpack-merge')
+
+// We can include Webpack plugins, through addons, that do
+// not need to run every time we are developing.
+// We will see an example when we set up 'Bundle Analyzer'
+const addons = (/* string | string[] */ addonsArg) => {
+  // Normalize array of addons (flatten)
+  let addons = [...[addonsArg]].filter(Boolean) // If addons is undefined, filter it out
+
+  return addons.map((addonName) => require(`./build-utils/addons/webpack.${addonName}.js`))
+}
+
+// 'env' will contain the environment variable from 'scripts'
+// section in 'package.json'.
+// console.log(env); => { env: 'dev' }
+module.exports = (env) => {
+  // We use 'buildValidations' to check for the 'env' flag
+  if (!env) {
```

Comment:
this is not really checking that env.dev exists and if appropriate webpack file exists, maybe just:

`module.exports = ({ env = 'dev' }) => {`
and
` const envConfig = require(`./build-utils/webpack.${env}.js`)`
that way don't really need the error check

Reply:
also forgot to mention, personal preference for me, is to use full word 'development', 'production'

Reply:
Yes, good point about the possibility of not `env` passed. 
And for me the naming in this case is not a big deal. Most times I agree to never user shortage version of names. But in this case is like using `i` for iteration. I'm changing to `development` and `production` anyway.

Reply:
Oh, I reverted the change on the file names because in VS Code it stopped showing as `webpack` types if the different name, so I think this is what is expected to be used for naming these files.